### PR TITLE
board: raspberrypi: Enable serial communication on Raspberry Pi 5

### PR DESCRIPTION
**Pull request by mypark42**
Enable serial communication through UART port between HDMI ports.

This is additional work on #70538.

UART base address, `0x107d001000`, came from dmesg in Ubuntu on Raspberry Pi 5 and from Raspberry Pi 5 linux dts.
UART baud rate was determined to be a widely used value of `115200` for convenience.
UART clock frequency, `44236800`, came from `/sys/kernel/debug/clk/uart-clock/clk_rate` in Ubuntu on Raspberry Pi 5.

### CODE REVIEW

File: drivers/gpio/gpio_brcmstb.c [hunk 1]
```diff
@@ -0,0 +1,180 @@
...
+		SET_BIT(reg, pin);                                                                 \
+		sys_write32(reg, addr);                                                            \
+	} while (0)
+#define CLR_BIT(reg, pin) reg &= ~((uint32_t)(1 << pin));
+#define CLR_BIT_RW(reg, addr, pin)                                                                 \
+	do {                                                                                       \
+		reg = sys_read32(addr);                                                            \
+		CLR_BIT(reg, pin);                                                                 \
+		sys_write32(reg, addr);                                                            \
+	} while (0)
```
Comment (carlocaione) [inline drivers/gpio/gpio_brcmstb.c:37]: Use the helpers in `sys_bitops.h` instead.

File: drivers/gpio/gpio_brcmstb.c [hunk 2]
```diff
@@ -0,0 +1,180 @@
...
+
+#define DEV_CFG(dev)  ((const struct gpio_brcmstb_config *)(dev)->config)
+#define DEV_DATA(dev) ((struct gpio_brcmstb_data *)(dev)->data)
```
Comment (carlocaione) [inline drivers/gpio/gpio_brcmstb.c:40]: Do not use `DEV_*` macros to access data and config.

**Review by carlocaione — CHANGES_REQUESTED** [review]

File: drivers/gpio/gpio_brcmstb.c (hunk 2 above)
Comment (42mypark) [reply drivers/gpio/gpio_brcmstb.c:26]: All uses of those macros have been removed. Should I also remove them? If so, using the `DEVICE_MMIO_NAMED_MAP` macro is inconvenient.

File: drivers/gpio/gpio_brcmstb.c (hunk 1 above)
Comment (42mypark) [reply drivers/gpio/gpio_brcmstb.c]: We replaced all macros with helper functions. Thanks.

**Comment by Lahyllas** [comment]
Since the author cannot maintain this PR anymore, I am going to deal with reviews or requests on this PR from now on.

**Comment by soburi** [comment]
@Lahyllas 

> Since the author @42mypark cannot maintain this PR anymore, I am going to deal with reviews or requests on this PR from now on.

In that case, I don't think you can push to this branch, so I think it's better to handle it with a PR that you can manage.
The difference is serial support only to #70538, so why not incorporate Serial support into #70538?

**Comment by Lahyllas** [comment]
> @Lahyllas
> 
> > Since the author @42mypark cannot maintain this PR anymore, I am going to deal with reviews or requests on this PR from now on.
> 
> In that case, I don't think you can push to this branch, so I think it's better to handle it with a PR that you can manage. The difference is serial support only to #70538, so why not incorporate Serial support into #70538?

I think that would be better. Thank you for your suggestion.

**Comment by soburi** [comment]
The same content has been merged with #70538, so I closed it.

**Pull request closed without merging**